presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1637,7 +1639,18 @@ core::PlanNodePtr VeloxQueryPlanConverterBase::toVeloxQueryPlan(
   }
   if (auto aggregation =
           std::dynamic_pointer_cast<const protocol::AggregationNode>(node)) {
-    return toVeloxQueryPlan(aggregation, tableWriteInfo, taskId);
+    bool isStreamingAggAllowed = true;
+    auto parent = aggregation->source;
+    while (parent.get()) {
+      if (auto joinNode =
+              std::dynamic_pointer_cast<protocol::JoinNode>(parent)) {
+        isStreamingAggAllowed = !joinNode->criteria.empty();
```

Comment:
If you aren't exiting the loop early, then you need to && this with isStreamingAggAllowed so you don't overwrite a false value with a true value (e.g. you have a join on top of a join, and only the top one is nestedLoopJoin)

Reply:
Edit: sorry misread the code, I see you are exiting early, but you should only exit if it's false. Otherwise you should continue.

Reply:
@rschlussel I am still working on this. The `source` needs to be moved to the PlanNode and I cannot edit the presto_protocol.h directly for that. I am still figuring out best way to check the ancestors.

Reply:
@rschlussel : This fix can be simplified since we only allow Project, Filter or Limit, DistinctLimit in between. In any other case streaming should be immediately returned false.

Think Karteek is changing to that.

But if this doesn't wrap up in the next round, we can do with never setting pre-grouping.